pipeline.py

```diff
@@ -180,318 +185,397 @@ def pipeline(
+                merge_system_user_message=final_eval_merge_system_user_message,
+                few_shots=final_eval_few_shots,
+                batch_size=final_eval_batch_size,
+            )
+            mount_pvc(
+                task=final_eval_task,
+                pvc_name=output_pvc_task.output,
+                mount_path="/output",
+            )
+            mount_pvc(
+                task=final_eval_task,
+                pvc_name=sdg_input_pvc_task.output,
+                mount_path="/input",
+            )
+            mount_pvc(
+                task=final_eval_task,
+                pvc_name=model_pvc_task.output,
+                mount_path="/model",
+            )
+
+            use_config_map_as_env(
+                final_eval_task,
+                JUDGE_CONFIG_MAP,
+                dict(endpoint="JUDGE_ENDPOINT", model="JUDGE_NAME"),
+            )
+
+            final_eval_task.set_env_variable("HOME", "/tmp")
+            final_eval_task.set_env_variable("HF_HOME", "/tmp")
+            set_image_pull_secrets(final_eval_task, [IMAGE_PULL_SECRET])
+
+            # uncomment if updating image with same tag
+            # set_image_pull_policy(final_eval_task, "Always")
+
+            use_secret_as_env(
+                final_eval_task, JUDGE_SECRET, {"api_key": "JUDGE_API_KEY"}
+            )
+
+            final_eval_task.after(run_mt_bench_task)
+            final_eval_task.set_accelerator_type("nvidia.com/gpu")
+            final_eval_task.set_accelerator_limit(1)
+
+            return final_eval_task
+
+        def outputs_to_artifacts():
+            output_model_task = pvc_to_model_op(
+                pvc_path="/output/phase_2/model/hf_format/candidate_model",
+            )
+            output_model_task.after(run_mt_bench_task)
+            output_model_task.set_caching_options(False)
+            mount_pvc(
+                task=output_model_task,
+                pvc_name=output_pvc_task.output,
+                mount_path="/output",
+            )
+
+            output_mt_bench_task = pvc_to_mt_bench_op(
+                pvc_path="/output/mt_bench_data.json",
+            )
+            output_mt_bench_task.after(run_mt_bench_task)
+            mount_pvc(
+                task=output_mt_bench_task,
+                pvc_name=output_pvc_task.output,
+                mount_path="/output",
+            )
+
+        # Pipelines
+
+        # some pre-filght image that checks all params and fails if something is wrong.
+
+        with dsl.If(sdg_only == True and train_only == True, name="Skip Condition"):
+            output_pvc_task, sdg_input_pvc_task, model_pvc_task = create_pvcs()
+            delete_pvcs(
+                output_pvc_task=output_pvc_task,
+                sdg_input_pvc_task=sdg_input_pvc_task,
+                model_pvc_task=model_pvc_task,
+                after=output_pvc_task,
+            )
+
+        with dsl.If(sdg_only == True and train_only == False, name="SDG Only"):
```

Comment:
Can we flip the toggles? It may be more intuitive that way... like remove the `*_only` meaning from it:

| Param names | Default value |
| --- | --- |
| `run_sdg` | True |
| `run_train` | True |

So it would work like this:

| `run_sdg` value | `run_train` value | Outcome |
| --- | --- | --- |
| True | True | Full run |
| True | False | Only SDG is run |
| False | True | Oly Training is run |
| False | False | Nothing happens |


WDYT?

Reply:
yes, that is much better 😄